Automattic/wp-calypso
# WPCOM: Remove `wpcom.undocumented()` from this world

#### Changes proposed in this Pull Request

This PR does the last cleanup to remove `wpcom.undocumented()` completely. It also removes some legacy references and unnecessary mocks. We're also renaming the `wpcom-undocumented` wrapper to `wpcom-oauth-wrapper`, because that's what the remaining functionality of that wrapper does.  

A good potential follow-up to this PR could be to call that wrapper only when the `oauth` feature flag is enabled, but if it's disabled, just use `wpcom-proxy-request` directly.

Fixes #58458.

#### Testing instructions

* Verify Calypso builds well.
* Smoke test Calypso and verify you can still perform requests properly. 
* Smoke test Jetpack Cloud and verify you can still perform requests properly.
* Go through the user signup flow and verify it still works well.
* Verify all tests pass.
